### PR TITLE
Merge develop into master

### PR DESCRIPTION
**Pull request by francoishamon**
@TotoGaz I think merging this PR is the last step before I can merge https://github.com/GEOSX/GEOSX/pull/1351 to synchronize GEOSX with the latest version of `PVTPackage`.

### CODE REVIEW

**Comment by TotoGaz** [comment]
@francoishamon I mostly already did the review when we merged into `master`, didn't I?

**Comment by francoishamon** [comment]
> @francoishamon I mostly already did the review when we merged into `master`, didn't I?

Yes i should be the same code that you have already reviewed (probably many times at this point).

File: PVTPackage/source/MultiphaseSystem/NegativeTwoPhaseMultiphaseSystem.cpp [hunk 1]
```diff
@@ -18,6 +18,10 @@
 
 #include "pvt/pvt.hpp"
 
+#if defined(GEOSX_USE_OPENMP)
+#include <omp.h>
+#include <chrono>
+#endif
```
Comment (TotoGaz) [inline PVTPackage/source/MultiphaseSystem/NegativeTwoPhaseMultiphaseSystem.cpp:24]: We cannot send this to `master`. Let's put some `PVTPACKAGE_USE_OPENMP` instead.
Or simply remove it.

File: PVTPackage/source/MultiphaseSystem/NegativeTwoPhaseMultiphaseSystem.hpp [hunk 2]
```diff
@@ -43,6 +43,16 @@ class NegativeTwoPhaseMultiphaseSystem : public CompositionalMultiphaseSystem
...
+  const std::vector< NegativeTwoPhaseFlashMultiphaseSystemProperties > & getBatchMultiphaseSystemProperties() const
+  {
+    return m_batchProps;
+  }
+#if defined(GEOSX_USE_OPENMP)
+  // Not very good, it's temporary...
+  void BatchUpdate( const std::vector< double > & pressure,
+                    const std::vector< double > & temperature,
+                    const std::vector< std::vector< double > > & feed );
+#endif
```
Comment (TotoGaz) [inline PVTPackage/source/MultiphaseSystem/NegativeTwoPhaseMultiphaseSystem.hpp:55]: Here too.

File: PVTPackage/source/MultiphaseSystem/NegativeTwoPhaseMultiphaseSystem.hpp [hunk 3]
```diff
@@ -43,6 +43,16 @@ class NegativeTwoPhaseMultiphaseSystem : public CompositionalMultiphaseSystem
                double temperature,
                std::vector< double > feed ) override;
 
+  const std::vector< NegativeTwoPhaseFlashMultiphaseSystemProperties > & getBatchMultiphaseSystemProperties() const
+  {
+    return m_batchProps;
+  }
```
Comment (TotoGaz) [inline PVTPackage/source/MultiphaseSystem/NegativeTwoPhaseMultiphaseSystem.hpp:49]: Hmm, this should be surrounded by `#if defined(GEOSX_USE_OPENMP)` too...
Probably a mistake

File: PVTPackage/source/MultiphaseSystem/NegativeTwoPhaseMultiphaseSystem.hpp [hunk 4]
```diff
@@ -54,6 +64,8 @@ class NegativeTwoPhaseMultiphaseSystem : public CompositionalMultiphaseSystem
   NegativeTwoPhaseFlash m_negativeTwoPhaseFlash;
 
   NegativeTwoPhaseFlashMultiphaseSystemProperties m_ntpfmsp;
+
+  std::vector< NegativeTwoPhaseFlashMultiphaseSystemProperties > m_batchProps;
```
Comment (TotoGaz) [inline PVTPackage/source/MultiphaseSystem/NegativeTwoPhaseMultiphaseSystem.hpp:68]: And this should not exist without openmp neither.

**Review by TotoGaz — COMMENTED** [review]
1) Can you check what's going on with BLT. I am surprised that we see some change.
2) We need to be careful with this `openmp` stuff, please have a look.
3) Maybe we want to completely remove the `tests` folder so it does not reach `GEOSX` and nobody cares about what's been done here. What's your point of view?

I can unprotect the `develop` branch (probably can you too) to make this duty easier.

File: PVTPackage/source/MultiphaseSystem/NegativeTwoPhaseMultiphaseSystem.hpp (hunk 4 above)
Comment (francoishamon) [reply PVTPackage/source/MultiphaseSystem/NegativeTwoPhaseMultiphaseSystem.hpp]: I removed this

File: PVTPackage/source/MultiphaseSystem/NegativeTwoPhaseMultiphaseSystem.hpp (hunk 3 above)
Comment (francoishamon) [reply PVTPackage/source/MultiphaseSystem/NegativeTwoPhaseMultiphaseSystem.hpp]: I removed this

File: PVTPackage/source/MultiphaseSystem/NegativeTwoPhaseMultiphaseSystem.hpp (hunk 2 above)
Comment (francoishamon) [reply PVTPackage/source/MultiphaseSystem/NegativeTwoPhaseMultiphaseSystem.hpp]: I removed this

File: PVTPackage/source/MultiphaseSystem/NegativeTwoPhaseMultiphaseSystem.cpp (hunk 1 above)
Comment (francoishamon) [reply PVTPackage/source/MultiphaseSystem/NegativeTwoPhaseMultiphaseSystem.cpp]: I removed this

File: PVTPackage/source/MultiphaseSystem/NegativeTwoPhaseMultiphaseSystem.cpp [hunk 5]
```diff
@@ -80,4 +80,38 @@ const pvt::MultiphaseSystemProperties & NegativeTwoPhaseMultiphaseSystem::getMul
...
+//    std::cout << omp_get_thread_num() << std::endl;
+//    m_negativeTwoPhaseFlash.computeEquilibrium( m_batchProps[i] );
+    computeEquilibriumAndDerivativesWithTemperature( m_negativeTwoPhaseFlash, m_batchProps[i] );
+  }
+// do something
+  auto t_end = std::chrono::high_resolution_clock::now();
+  double elapsed_time_ms = std::chrono::duration< double, std::milli >( t_end - t_start ).count();
+  std::cout << "TIME (ms) = " << elapsed_time_ms << std::endl;
+}
+#endif
```
Comment (francoishamon) [inline PVTPackage/source/MultiphaseSystem/NegativeTwoPhaseMultiphaseSystem.cpp:116]: I forgot to remove this.

Comment (francoishamon) [reply PVTPackage/source/MultiphaseSystem/NegativeTwoPhaseMultiphaseSystem.cpp]: It is removed now

**Comment by francoishamon** [comment]
@TotoGaz I think I removed at the OpenMP stuff from `develop`. The changes caused by merging `develop` into `master` come from:
- Some changes in BlackOil and DeadOil models (harmless, no longer used in GEOSX)
- Josh's fix (useful in GEOSX)
- The tests that you added
- A BLT update

Do you think I should update BLT to the same version as in GEOSX?

**Comment by TotoGaz** [comment]
It's not mandatory to have the same BTL, but it can be better. I wouldn't worry if I were you (other projects use others BLT so...)

**Comment by francoishamon** [comment]
@TotoGaz if that is ok with you, we can merge this PR and then update GEOSX tomorrow

**Review by TotoGaz — APPROVED** [review]

**Pull request merged**